torch_xla_py/xla_dist.py

```diff
@@ -402,3 +409,363 @@ def get_cluster(self):
     cluster.validate()
     return cluster
 
+
+class DistributedExecutor(object):
+
+  SCRIPT_PATH_TMPL = '/tmp/{pid}/dist_training_ptxla_{worker}.sh'
+  MASTER_IDX = 0
+  MESH_SERVICE_PORT = 8477  # Use single port to disallow concurrent runs
+  DIST_ENV_VARS = [
+      'XRT_TPU_CONFIG', 'XRT_LOCAL_WORKER', 'XRT_MESH_SERVICE_ADDRESS'
+  ]
+  DEFAULT_CONTAINER_NAME = 'pytorchtpudistrunner'
+  DEFAULT_USER_NAME = 'pytorchtpudistrunner'
+
+  @staticmethod
+  def _concat_cmd_lst(cmd_lst, delimiter=' '):
+    return delimiter.join(cmd_lst)
```

Comment:
Can we get here with strings in `cmd_list` which are already delimited?

Reply:
This shouldn't happen the way things are right now and even it they were I don't think it'd make things break, but I have added a check for this. If already delimited we don't delimit again.